File: tool/table/server/cards_dungeon.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


cards_dungeon_map = {};
cards_dungeon_map[1] = {"lv":1,"min":6,"max":16,"equip_group":"1,2","equip_min":2,"equip_max":3,"spell_group":"7.0","spell_min":1,"spell_max":3,"trap_group":"8.0","trap_min":1,"trap_max":2,"monster_group":"3,4,5,6",};


class Cards_dungeon:
	def __init__(self, key):
		config = cards_dungeon_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Cards_dungeon(key):
		config = cards_dungeon_map.get(key);
		if not config:
			return
		return Cards_dungeon(key)

File: tool/table/client/cards_dungeon.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let cards_dungeon_map = null;
export function cards_dungeon_map_init(config_obj:Object):void{
	cards_dungeon_map = config_obj["cards_dungeon_map"];
}


export class Cards_dungeon extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = cards_dungeon_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Cards_dungeon(key){
		if(Cards_dungeon.m_static_map.hasOwnProperty(key) == false){
			Cards_dungeon.m_static_map[key] = Cards_dungeon.create_Cards_dungeon(key);
		}
		return Cards_dungeon.m_static_map[key];	
	}
	public static create_Cards_dungeon(key){
		if(cards_dungeon_map.hasOwnProperty(key)){
			return new Cards_dungeon(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return cards_dungeon_map;
	}
}
}